tests/config-override.test.sh

```
@@ -0,0 +1,485 @@
+# _isolate runs a snippet in a clean PWD with a fresh HOME and an empty
+# config file (caller may overwrite). Sentinels reset by running each
+# snippet in a sub-shell.
+_isolate() {
+  local body="$1"
+  local sandbox stderr
+  sandbox="$(mktemp -d)"
+  stderr="$(mktemp)"
+  mkdir -p "$sandbox/.claude"
+  : > "$sandbox/.claude/uberdev.local.md"
+  (
+    cd "$sandbox"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)